vnext/Microsoft.ReactNative/Views/TextViewManager.cpp

```diff
@@ -288,6 +288,22 @@ bool TextViewManager::UpdateProperty(
       node->m_backgroundColor = OptionalColorFrom(propertyValue);
       node->RecalculateTextHighlighters();
     }
+  } else if (propertyName == "accessibilityRole") {
+    if (propertyValue.Type() == winrt::Microsoft::ReactNative::JSValueType::String) {
+      const std::string &role = propertyValue.AsString();
+      auto value = asHstring(propertyValue);
+      auto boxedValue = winrt::Windows::Foundation::PropertyValue::CreateString(value);
+
+      textBlock.SetValue(winrt::AutomationProperties::LocalizedControlTypeProperty(), boxedValue);
+      if (role == "header") {
+        xaml::Automation::AutomationProperties::SetHeadingLevel(
+            textBlock, winrt::Peers::AutomationHeadingLevel::Level2);
```

Comment:
Is there a reason why you chose heading level 2 as the default as opposed to level 1?

Reply:
@ksiler The team currently blocked by this issue was needing Level2 headings specifically, that's really the main reason why. Also, [looks like Adaptive Cards has chosen level 2 as the default](https://github.com/microsoft/AdaptiveCards/blob/main/specs/DesignDiscussions/TextHeadings.md) -- I'm checking in with them about that.